### DISCUSSION TITLE
BooleanColumn: is it possible to use a closure to process false/true situation based on a none boolean (string) data?

### DISCUSSION THREAD

**Discussion by ehs4nm** [filamentphp/filament#1811] [Help] [answered]
I have a relationship for my model, question & answer, questions are asked but may not be answered and the response would be null.
I am using a TextColumn for this now but i would be nice to have features like what BooleanColumn has, icons or colors. I am not sure that if it worth it but just my two cents idea to share.

thanks for the work you do guys. Awesome project.

**Comment by ryangjchandler** [comment, marked as answer]
You can add the following to your column:

```php
->getStateUsing(function ($record) {
	return (bool) $record->my_column;
})
```

**Comment by DavidDevGt** [comment]
in filament use this->https://docs.laravel-filament.cn/docs/tables/columns/icon/
